flink-runtime/src/main/java/org/apache/flink/runtime/rpc/resourcemanager/ResourceManager.java

```diff
@@ -53,14 +56,23 @@
  */
 public class ResourceManager extends RpcEndpoint<ResourceManagerGateway> {
 	private final Map<JobMasterGateway, InstanceID> jobMasterGateways;
+
+	/** ResourceID and TaskExecutorGateway mapping relationship of registered taskExecutors */
+	private final Map<ResourceID, TaskExecutorGateway>  startedTaskExecutorGateways;
+
+	/** TaskExecutorGateway and InstanceId mapping relationship of registered taskExecutors */
+	private final Map<TaskExecutorGateway, InstanceID> taskExecutorGateways;
```

Comment:
Wouldn't it make sense to group the `TaskExecutorGateway` and the `InstanceID` into a `TaskExecutorRegistration` class which is stored under the resource ID? Then we would get rid of a lookup when accessing the `InstanceID` given the resource ID.